### PR TITLE
Create config for pre-commit and autoformatters+flake8

### PR DESCRIPTION
**Pull request by ianhi**
Once pre-commit.ci is fully enabled then the pot should push the style fixes to this PR as a single commit. https://github.com/matplotlib/ipympl/issues/360

### CODE REVIEW

**Comment by ianhi** [comment]
Not enabled yet. But once enabled a close + reopen should trigger it

**Comment by martinRenou** [comment]
It should be enabled now

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - id: isort
+  - repo: https://github.com/psf/black
+    rev: 21.5b2
+    hooks:
+      - id: black
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.19.0
+    hooks:
+      - id: pyupgrade
+        args: [--py37-plus]
```
Comment (martinRenou) [inline .pre-commit-config.yaml:30]: Is it possible to do the same for eslint and prettier?

Comment (ianhi) [reply .pre-commit-config.yaml:30]: seems like it: https://github.com/pre-commit/mirrors-eslint and https://github.com/pre-commit/mirrors-prettier

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    hooks:
+      - id: prettier
+        types: [javascript]
+
+  - repo: https://github.com/pre-commit/mirrors-eslint
+    rev: 'v7.32.0'
+    hooks:
+    - id: eslint
+      additional_dependencies:
+        - "@typescript-eslint/eslint-plugin"
```
Comment (ianhi) [inline .pre-commit-config.yaml:42]: I think I need to add more things here before merging. not sure i got this right yet.

**Comment by ianhi** [comment]
Only took 11 force pushes :) but we got there

**Comment by martinRenou** [comment]
Amazing. Thanks!

**Pull request merged**